### ISSUE TITLE
Changing ActiveProfile: Collection changed during LateUpdate

### ISSUE THREAD

**Issue by davidkline-ms** [microsoft/MixedRealityToolkit-Unity#6042]
## Describe the bug

When swapping the MixedRealityToolkit.ActiveProfile, the following InvalidOperationException (collection changed) occurs during LateUpdate.

```
Create a new scene and configure for the MixedRealityToolkit
Add a button to the scene and apply / configure the included script
Enter play mode and expand the MixedRealityPlayspace object in the hierarchy
Click the button to toggle between profiles
```

## To reproduce

1. Create a new scene and configure for the MixedRealityToolkit
2. Add a button to the scene and apply / configure the included script
3. Enter play mode and expand the MixedRealityPlayspace object in the hierarchy
4. Click the button to toggle between profiles

## Expected behavior

The collection modification either does not occur, or the enumeration is aborted during [Late]Update when it is modified.

## Your Setup (please complete the following information)

- Unity Version 2018.4.9f1
- MRTK Version 2.0.0 + fixes that will be in 2.1.0

## SwapActiveProfile.cs

```c#
using Microsoft.MixedReality.Toolkit;
using UnityEngine;

public class SwapActiveProfile : MonoBehaviour
{
    public MixedRealityToolkitConfigurationProfile profile1;
    public MixedRealityToolkitConfigurationProfile profile2;

    void Start()
    { }

    public void Swap()
    {
        MixedRealityToolkit.Instance.ActiveProfile = (MixedRealityToolkit.Instance.ActiveProfile == profile1) ? profile2 : profile1;
    }
}
```

**Comment by davidkline-ms** [comment]
This _may_ be related to a recent change to the MixedRealityServiceRegistry.GetAllServices(). Investigating...

**Comment by davidkline-ms** [comment]
Fixed in personal fork as part of #6043
